website/docs/r/dataproc_cluster.html.markdown

```diff
@@ -0,0 +1,202 @@
+---
+layout: "google"
+page_title: "Google: google_dataproc_cluster"
+sidebar_current: "docs-google-dataproc-cluster"
+description: |-
+  Manages a Cloud Dataproc cluster resource.
+---
+
+# google\_dataproc\_cluster
+
+Manages a Cloud Dataproc cluster resource within GCE. For more information see
+[the official dataproc documentation](https://cloud.google.com/dataproc/).
+
+
+!> **Warning:** Due to limitations of the API, all arguments except
+`labels`,`worker_config.num_workers` and `worker_config.preemptible_num_workers` are non-updateable. Changing any will cause recreation of the
+whole cluster!
+
+## Example usage
+
+```hcl
+resource "google_dataproc_cluster" "mycluster" {
+    name   = "dproc-cluster-unique-name"
+    region = "us-central1"
+
+    master_config {
+        num_masters       = 1
+        machine_type      = "n1-standard-1"
+        boot_disk_size_gb = 10
+        num_local_ssds    = 1
+    }
+
+    worker_config {
+    	num_workers             = 2
+        machine_type            = "n1-standard-1"
+        boot_disk_size_gb       = 10
+        num_local_ssds          = 1
+
+        preemptible_num_workers       = 1
+        preemptible_boot_disk_size_gb = 10
+    }
+
+    initialization_action_timeout_sec = 500
+    initialization_actions = [
+       "gs://dataproc-initialization-actions/stackdriver/stackdriver.sh"
+    ]
+    
+    labels {
+      foo = "bar"
+    }
+    
+    tags = ["foo", "bar"]
+
+  }
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) The name of the cluster, unique within the project and
+    zone.
+
+- - -
+
+* `region` - (Optional) The region that the cluster and associated nodes will be created in.
+   Defaults to `global`.
+
+* `zone` - (Optional) The GCP zone where your data is stored and used (i.e. where
+    the master and the worker nodes will be created in). If region is set to 'global'
+    then `zone` IS mandatory, otherwise GCP is able to make use of [Auto Zone Placement](https://cloud.google.com/dataproc/docs/concepts/auto-zone)
+    to determine this automatically for you.
+    Note: This setting additionally determines and restricts
+    which computing resources are available for use with other config such as
+    `master_config.machine_type` and `worker_config.machine_type`.
+
+* `staging_bucket` - (Optional) The Cloud Storage staging bucket used to stage files,
+   such as Hadoop jars, between client machines and the cluster.
+   Note: If you don't explicitly specify a `staging_bucket`
+   then GCP will auto create / assign one for you, however you are NOT guaranteed to get
```

Comment:
nit: rephrase to `then GCP will auto create / assign one for you. However, you are not guaranteed`